File: leetcode/editor/cn/434_字符串中的单词数_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when434_thenSuccess(t *testing.T) {
	assert.Equal(t, 5, countSegments("Hello, my name is John"))
	assert.Equal(t, 5, countSegments("Hello, my name is John  "))
}

//leetcode submit region begin(Prohibit modification and deletion)
func countSegments(s string) int {
	ans := 0
	word := false
	for _, char := range s {
		if ' ' == char {
			if word {
				ans += 1
				word = false
			}
		} else {
			word = true
		}
	}
	if word {
		ans += 1
	}
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/cn/434_字符串中的单词数_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertTrue(True, "ShouldBeTrue")


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def countSegments(self, s: str) -> int:
        ans = 0
        for i, c in enumerate(s):
            if c != ' ' and (i == 0 or s[i - 1] == ' '):
                ans += 1
        return ans


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()
